usefulUtilities/botPlayer/assignmentClientManager.js

```diff
@@ -0,0 +1,248 @@
+"use strict";
+
+//
+//  Bot Player
+//  assignmentClientManager.js
+//  Created by Milad Nazeri on 2019-06-06
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+
+(function() {
+
+    // *************************************
+    // START UTILITY FUNCTIONS
+    // *************************************
+    // #region UTILITY FUNCTIONS   
+    
+
+    // Start playing sequence to fill players with bots
+    var AC_AVAILABILITY_CHECK_MS = 1000;
+    function startSequence() {
+        // Check to see how many bots are needed
+        if (botCount >= botsFound + 1) {
+            return;
+        }
+
+        if (botCount < availableAssignmentClientPlayers.length) {
+            var player = availableAssignmentClientPlayers[botCount];
+            player.play();
+            botCount++;
+
+            if (botCount >= botsFound + 1) {
+                return;
+            }
+        }
+
+        Script.setTimeout(function() {
+            startSequence();
+        }, AC_AVAILABILITY_CHECK_MS);
+    }
+
+
+    // Searching through the s3 bucket to grab which recordings are valid and stop as soon as we get an error
+    // Synchronous version of require
+    var BASE_PATH = "https://hifi-content.s3.amazonaws.com/howell/bots/usertesting/";
+    var MAX_BOTS_TO_TRY = 100;
+    var request = Script.require("./requestSync.js").request;
+    var botsFound = 0;
+    function populateRecordingList(){
+        for (var i = 1; i < MAX_BOTS_TO_TRY; i++) {
+            var botRecordingFound = true;
+            var currentBotUrl = BASE_PATH + "AVATAR_TEST" + i + ".hfr";
+            request(currentBotUrl, function(error){
+                if (error) {
+                    botRecordingFound = false;
+                } else {
+                    botsFound++;
+                    botList.push(currentBotUrl);
+                }
+            });
+            if (!botRecordingFound) {
```

Comment:
doesn't this have to be inside the `request()` ?
EDIT: nevermind, request is synchronous. could you please change the name of the function to something like `requestSync` so that it's more clear?

Reply:
also, what's supposed to happen in the `break` case?

Reply:
First point - good idea, will change

second one - since this is sync, it breaks out of searching for more avatars needed and then the next function called starts the sequence